Listening to FCM notification when app is in killed state

I am using latest version of NotificationHub(1.0.1). To receive to notifications when application is in killed state I need to start the azure SDK in a custom class that extends Application.class in android. I get exception in the logs when call the below method
`NotificationHub.start(this.getApplication(), BuildConfig.hubName, BuildConfig.hubListenConnectionString);`

Error message : Caused by: java.lang.IllegalStateException: Not allowed to start service Intent { cmp=com.cegid.flow/com.microsoft.windowsazure.messaging.notificationhubs.FirebaseReceiver }: app is in background uid UidRecord{70e259d u0a846 RCVR idle change:uncached procs:1 seq(0,0,0)}


In Android Oreo and above devices we cannot start a service from background. If I catch this exception then I can receive the notification in the Notification listener. Is there any other approach to register for notification when app is killed.

If I do not call NotificationHub.start() method in the custom Application class then I get below exception.

Error message: java.lang.RuntimeException: Unable to create service com.microsoft.windowsazure.messaging.notificationhubs.FirebaseReceiver: java.lang.NullPointerException: Attempt to invoke virtual method 'java.lang.String com.microsoft.windowsazure.messaging.notificationhubs.PushChannelVisitor.getPushChannel()' on a null object reference.

 - OS: Android 9
 - IDE : Android Studio 4.0.1
 - 1.0.1

--- a/notification-hubs-sdk/src/main/java/com/microsoft/windowsazure/messaging/notificationhubs/NotificationHub.java
+++ b/notification-hubs-sdk/src/main/java/com/microsoft/windowsazure/messaging/notificationhubs/NotificationHub.java
@@ -115,9 +115,6 @@ public final class NotificationHub {
         instance.mPushChannelVisitor = new PushChannelVisitor(instance.mApplication);
         instance.useInstanceVisitor(instance.mPushChannelVisitor);
 
-        Intent i =  new Intent(application, FirebaseReceiver.class);
-        application.startService(i);
-
         // Why is this done here instead of being in the manifest like everything else?
         // BroadcastReceivers are special, and starting in Android 8.0 the ability to start them
         // from the manifest was removed. See documentation from Google here:

